### PR TITLE
Support TOML v1.0.0 syntax in `pyproject.toml`

### PR DESCRIPTION
**Pull request by hukkin**
Closes https://github.com/pytest-dev/pytest/issues/8789

### CODE REVIEW

File: changelog/8789.feature.rst
Comment (The-Compiler) [inline changelog/8789.feature.rst:1]: This should point out that the `toml` dependency got replaced by `tomli` - something e.g. distribution packagers could easily miss and definitely should know about!

File: setup.cfg
Comment (The-Compiler) [inline setup.cfg:50]: I don't think the `>=1.0.0` is needed here, given that there never was a < 1.0.0 release. Not sure about the `<2.0.0` part. We don't typically add upper bounds for dependencies, and I guess the tiny bit of API we need will probably stay the same across future versions.

File: src/_pytest/config/findpaths.py
Comment (The-Compiler) [inline src/_pytest/config/findpaths.py:69]: Not directly related to your PR, but more of a question to @nicoddemus: I wonder why we don't handle `tomli.TOMLDecodeError` here and turn it into an `UsageError`, like we do with `_parse_ini_config` above?

**Review by The-Compiler — CHANGES_REQUESTED** [review]
Thanks!

I wonder what our policy on dependency changes has been so far? Can we do this in a minor release, or do we typically introduce new dependencies in major releases only? If the latter, I guess we could squeeze this into 7.0 as it's pretty trivial.

File: src/_pytest/config/findpaths.py
Comment (RonnyPfannschmidt) [reply src/_pytest/config/findpaths.py:69]: Good idea, i think it should be done

Its up to @hukkin whether this is ok for this pr or if it will be a follow-up

**Review by RonnyPfannschmidt — APPROVED** [review]
I like it, let's land it in 7.0

File: setup.cfg
Comment (hukkin) [reply setup.cfg:50]: There are [<1.0.0 releases](https://pypi.org/project/tomli/#history). I had this same discussion [with Black maintainers recently](https://github.com/psf/black/pull/2301#discussion_r647748572) :smile: . You'll find my thoughts in that thread.

If you tell me exactly what constraint you want I'll change it to that of course (or you can just push to this branch).

File: src/_pytest/config/findpaths.py
Comment (hukkin) [reply src/_pytest/config/findpaths.py]: I made the change (as it's a very small change). Is this something we want a test for btw?

File: changelog/8789.feature.rst
Comment (hukkin) [reply changelog/8789.feature.rst]: :+1: Please have a look if the message now looks better to you

File: setup.cfg
Comment (The-Compiler) [reply setup.cfg:50]: Huh, I thought I had looked at PyPI, but apparently I messed up somehow. Ok, seems fine to me either way, let's see what others think.

**Review by The-Compiler — APPROVED** [review]

File: src/_pytest/config/findpaths.py
Comment (hukkin) [reply src/_pytest/config/findpaths.py]: >  Is this something we want a test for btw?

Well I thought it couldn't do any harm, so added a test for the exception type.

File: setup.cfg
Comment (nicoddemus) [reply setup.cfg:50]: I'm also not inclined to not having the upper pinning in there, because I don't consider `pytest` a user facing application, but a framework, so our dependencies need to be compatible downstream with user's code.

But having said that, I understand the points @hukkin made in that thread, and given he is the main author, his opinion here carries some weight. Also, seems we might never even see a 2.0 version. 😁 

In summary: -0 on the upper pinning.

File: src/_pytest/config/findpaths.py
Comment (nicoddemus) [reply src/_pytest/config/findpaths.py]: > Not directly related to your PR, but more of a question to @nicoddemus: I wonder why we don't handle tomli.TOMLDecodeError here and turn it into an UsageError, like we do with _parse_ini_config above?

Probably it was an oversight only, good catch!

**Review by nicoddemus — APPROVED** [review]
Thanks @hukkin!

**Comment by nicoddemus** [comment]
Should have squashed that, my bad. 😕

File: setup.cfg
Comment (asottile) [reply setup.cfg:50]: I'll put my -1 on the upper pinning, please remove it

Comment (nicoddemus) [reply setup.cfg:50]: Oh sorry, had enough approvals that I thought it would be OK to merge. 👍

**Pull request merged**
